File: .agents/skills/the-standard-reacttypescript-files/examples/good/example_good_file_naming.ts
// ---
// skill: the-standard-reacttypescript-files
// type: example
// source-section: "1. Project Structure — File Rules"
// demonstrates: "tsr-files-001, tsr-files-002, tsr-files-003, tsr-files-004"
// ---

// ✅ GOOD file naming — each file has one role, correct extension, descriptive name.

// src/brokers/apis/patientApiBroker.ts       — broker (.ts, role in name)
// src/brokers/apis/iPatientApiBroker.ts      — broker interface (.ts, i-prefix)
// src/services/foundations/patients/
//   patientService.ts                        — foundation service (.ts)
//   iPatientService.ts                       — service interface (.ts)
//   patientService.validations.ts            — validation partial (.ts)
//   patientService.exceptions.ts             — exception partial (.ts)
// src/services/views/dashboard/
//   dashboardViewService.ts                  — view service (.ts)
//   iDashboardViewService.ts                 — view service interface (.ts)
// src/pages/dashboard/
//   DashboardPage.tsx                        — page (.tsx, PascalCase, Page suffix)
//   useDashboardPage.ts                      — page hook (.ts, use-prefix)
// src/components/patients/
//   PatientSummaryCard.tsx                   — component (.tsx, PascalCase, Purpose in name)
// src/models/foundations/patients/
//   Patient.ts                               — foundation model (.ts)
// src/models/views/dashboard/
//   DashboardView.ts                         — view model (.ts)

File: .agents/skills/the-standard-csharp-files/examples/good/example_good_file_naming.cs
// ---
// skill: the-standard-csharp-files
// type: example
// source-section: "0. Files — 0.0 Naming, 0.1 Partial Class Files"
// demonstrates: "tsc-csharp-files-001, tsc-csharp-files-002, tsc-csharp-files-003"
// ---

// ✅ GOOD: Correct PascalCase file names and correct partial class dot-notation.

// File: Student.cs
public class Student { }

// File: StudentService.cs
public partial class StudentService : IStudentService { }

// File: StudentService.Validations.cs
public partial class StudentService
{
    private void ValidateStudentOnAdd(Student student) { }
}

// File: StudentService.Validations.Add.cs
public partial class StudentService
{
    private void ValidateStudentId(Guid studentId) { }
}

// File: StudentService.Exceptions.cs
public partial class StudentService
{
    private delegate ValueTask<Student> ReturningStudentFunction();

    private async ValueTask<Student> TryCatch(ReturningStudentFunction returningStudentFunction)
    {
        try
        {
            return await returningStudentFunction();
        }
        catch (NullStudentException nullStudentException)
        {
            throw new StudentValidationException(nullStudentException);
        }
    }
}
